### ISSUE TITLE
Top level async

### ISSUE THREAD

**Issue by michaelwclark** [wallabyjs/quokka#345]
### Issue description or question
I'd love to see quokka handle async/await at top level. For quick prototyping this would be awesome! 
### Sample code

```javascript
const test = async a => Promise.resolve(a)

console.log(test(4))
//Be awesome if we could.....
//console.log(await test(4)) 

//instead have to wrap async stuffs:
const getTest = async () => {
  const result = await test(4)
  console.log(result)
}

getTest().then()
```

### Quokka.js Console Output

```
​​​​Quokka 'Untitled-3.js' (node: v10.5.0)​​​​

Promise {} ​​​​​at ​​​test(4)​​​ ​quokka.js:3:0​

4 ​​​​​at ​​​result​​​ ​quokka.js:10:2​
```

**Comment by smcenlly** [comment]
Thanks for the feature request. We'll need to do some investigation to see if there are any side effects, but agree this would be a nice feature.

**Comment by ArtemGovorov** [comment]
While not exactly the same as the originally requested feature, Quokka [live comments](https://quokkajs.com/docs/#comments) (unlike `console.log`) support logging promise resolved values, so for your example you may do:

```
const test = async a => Promise.resolve(a)

test(4)//?
```

<img width="385" alt="Screen Shot 2019-03-23 at 9 09 39 am" src="https://user-images.githubusercontent.com/979966/54857507-68581600-4d4b-11e9-91d6-55db961b4b81.png">

**Comment by ArtemGovorov** [comment]
The feature is [implemented](https://twitter.com/wallabyjs/status/1111488427931758592) in the latest extension version.

**Comment by timarandras** [comment]
@ArtemGovorov While the simple top-level await is working, this is not:
```javascript
async function* asyncGen() {
  yield Promise.resolve(1);
}

for await (const ret of asyncGen()) {
  console.log(ret);
}
```
```
Failed to instrument quokka.js 
  3 | } 
  4 |  
> 5 | for await (const ret of asyncGen()) { 
    |     ^ SyntaxError: Unexpected token (5:4) 
  6 |   console.log(ret); 
  7 | }
```
Wrapping in an async funciton works tough:
```javascript
async function f() {
  for await (const ret of asyncGen()) {
    console.log(ret);
  }
}

f()
```
What can we do?

**Comment by smcenlly** [comment]
@timarandras - thanks for raising the issue. We've updated our instrumentation code to handle this case. Please update to the latest version of Quokka for your editor to get this fix.

**Comment by timarandras** [comment]
Works! Thank you.

**Comment by thunderkid** [comment]
This top-level await feature seems to work only in js files, and not in typescript files, where I still get the old `await is only allowed inside an async function` error. Is that correct? Any plans to allow in typescript too?

**Comment by smcenlly** [comment]
Unfortunately we can't support top-level await for TypeScript until it is supported by the TypeScript compiler. You may subscribe to the [TypeScript issue](https://github.com/microsoft/TypeScript/issues/25988) if you are interested in updates.

**Comment by SwiftWinds** [comment]
Top-level await was shipped in TypeScript 3.8. However, it still doesn't work in Quokka:

<img width="649" alt="image" src="https://user-images.githubusercontent.com/12981958/142174120-c96d67af-a340-4118-abed-65312496903a.png">

I'm using Webstorm 2021.2.3 w/ Quokka v1.0.317

**Comment by ArtemGovorov** [comment]
@SwiftWinds In order to make it work in Typescript, you need to configure your Quokka to [use native ESM](https://quokkajs.com/docs/configuration.html#es-modules-and-top-level-await), so that your Quokka file runs as a module (as the error message suggests).
